src/main/java/com/marginallyclever/makelangelo/select/SelectReadOnlyText.java

```diff
@@ -2,56 +2,83 @@
 
 import java.awt.BorderLayout;
 import java.awt.Desktop;
-import java.awt.event.MouseAdapter;
-import java.awt.event.MouseEvent;
 import java.io.IOException;
 import java.net.URI;
 import java.net.URISyntaxException;
+import javax.swing.JEditorPane;
 
-import javax.swing.JLabel;
+import javax.swing.ToolTipManager;
+import javax.swing.event.HyperlinkEvent;
+import javax.swing.event.HyperlinkListener;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
- * Read only HTML-rich text that can display AT MOST one clickable link.
- * @author Dan Royer
+ * Read only JEditorPane with an HyperLinckListener for text/html contening or not multiple html link.
+ * With clickable link (Desktop.BROWSE) and ToolTips when hovered over .
+ * 
+ * @author Dan Royer (and modified by PPAC37)
  * @since 7.24.0
  */
 public class SelectReadOnlyText extends Select {
 	/**
 	 * 
 	 */
-	private static final long serialVersionUID = -8918068053490064344L;
-	private static final String A_HREF = "<a href=";
-	private static final String HREF_CLOSED = ">";
+	private static final long serialVersionUID = -8918068053490064344L;// TODO to regenerat ?
 	
-	private JLabel label;
+	private static final Logger logger = LoggerFactory.getLogger(SelectReadOnlyText.class);
+		
+	private JEditorPane jEdPane;
 	
 	public SelectReadOnlyText(String internalName,String labelKey) {
-		super(internalName);
-		
-		label = new JLabel("<html>"+ labelKey+"</html>",JLabel.LEADING);
-		label.addMouseListener(new MouseAdapter() {
+		super(internalName);		
+		jEdPane = createJEditorPaneWithHyperlinkListenerAndToolTipsForDesktopBrowse("<html>"+ labelKey+"</html>");
+		this.add(jEdPane,BorderLayout.CENTER);
+	}
+	
+	/**
+	 * Create a JEditorPane not editable for text/html contente, with an HyperLinkListener to Desktop Browse (when clicked) and show a ToolTips with the URL hovered.
+	 * @param sToSetAsTextToTheHtmlEditorPane
+	 * @return 
+	 */
+	public static JEditorPane createJEditorPaneWithHyperlinkListenerAndToolTipsForDesktopBrowse(String sToSetAsTextToTheHtmlEditorPane) {
+		JEditorPane createdJEditorPane =  new JEditorPane();
+		createdJEditorPane.setEditable(false);
+		createdJEditorPane.setOpaque(false);
+		createdJEditorPane.setContentType("text/html");
+		createdJEditorPane.setText(sToSetAsTextToTheHtmlEditorPane);
+		createdJEditorPane.addHyperlinkListener(new HyperlinkListener() {
```

Comment:
```
createdJEditorPane.addHyperlinkListener( (hyperlinkEvent)->{
```
maybe?

Reply:
Thank you I too often forget the lambda expressions. I have to revisit my habits.